modules/task/cherami/cherami_test.go

```diff
@@ -261,13 +268,70 @@ func TestStartBackendOpenPublisherError(t *testing.T) {
 func TestStartBackendOpenConsumerError(t *testing.T) {
 	m := newMock()
 	defer m.AssertExpectations(t)
-	bknd := createNewBackend(t, m)
+	bknd := createNewBackend(t, m, _host)
 	errStr := "consume error"
 	_, err := startBackend(t, m, bknd, nil, errors.New(errStr))
 	assert.False(t, bknd.(*Backend).isRunning())
 	assert.Contains(t, err.Error(), errStr)
 }
 
+func TestEncodingErrors(t *testing.T) {
+	m := newMock()
+	defer m.AssertExpectations(t)
+	testArgs := []struct {
+		nackError    error
+		expectedLogs map[string]int
+	}{
+		{nil, map[string]int{"extract error": 1}},
+		{errors.New("nack error"), map[string]int{"extract error": 1, "nack error": 1}},
+	}
+	for _, testArg := range testArgs {
+		tracer := &tracing.ErrorTracer{Tracer: opentracing.NoopTracer{}}
+		zapLogger, buf := testutils.GetLockedInMemoryLogger()
+		defer ulog.SetLogger(zapLogger)()
+		host := service.NopHostConfigured(auth.NopClient, zapLogger, tracer)
+
+		bknd := createNewBackend(t, m, host)
+		cBknd := bknd.(*Backend)
+		m.Delivery.On("GetMessage").Return(
+			&cherami_gen.ConsumerMessage{
+				Payload: &cherami_gen.PutMessage{
+					Data:        _publishMsg,
+					UserContext: map[string]string{_ctxKey: ""},
+				},
+			},
+		)
+		m.Delivery.On("Nack").Return(testArg.nackError).Once()
```

Comment:
This is a method name, it comes from mock library to specify what to return on the "Nack" method call.